### DISCUSSION TITLE
[my usage experience] Receiving 3M requests per month on pocketbase

### DISCUSSION THREAD

**Discussion by erickythierry** [pocketbase/pocketbase#4506] [Show and tell]
I'm just sharing some metrics from my production instance in a small project I have.
I am somewhat impressed with the performance and quality of pocketbase, in my opinion it is ready for production use (for small to medium sized projects of course).

### some useful prints:

![logs](https://github.com/pocketbase/pocketbase/assets/11170366/3b5336f0-7bfc-42c7-87b1-01f47842bee1)
_my pocketbase instance receives an average of 100k requests per day (much of this comes from my own backend)_

#

![size](https://github.com/pocketbase/pocketbase/assets/11170366/a03bf6cf-d4a5-432c-84ee-1ec44c1f2cef)
_My DB is small, so I believe this influences the use of RAM.
I store a few thousand records, but these records are read and updated constantly._

#

![status](https://github.com/pocketbase/pocketbase/assets/11170366/03a94b3f-c26f-48ed-98ab-e6844497d08e)
_and this is the resource usage in the 1 day running docker container.
very efficient._

#

- I've been using pocketbase as the main DB for a few months in my project, I have just over 1400 users and it's working very well.
- I have around 3 million to 3.2 million requests to the database per month, but the vast majority are for reading and updating data
- I realized that as the logs grow, pocketbase's ram consumption grows too, so I limited log retention to just 1 day. Before, it used 7 days of retention, so the pocketbase reached 700-800MB of RAM used (which I still think is acceptable).
- I also noticed that writing data to disk is quite intense (reaching several tens of GB per month), I believe it is related to the logs.

## Well that's. It was a pleasant surprise to find this project, it solved my problems with DB incredibly easily, I hope the project continues to evolve, as it has a lot of potential.

**Comment by ganigeorgiev** [comment]
_Thanks for sharing! Just some side-notes:_

- The reported RAM usage may not necessary represent the current actual memory used by the process. The Go GC regularly notifies the OS for unused memory but it is up to the OS to decide whether to reclaim it or leave it (because the system has plenty to spare, there is no other competing process or some other reason). If you are operating in a memory constrained environment, you can define the [`GOMEMLIMIT=500MiB`](https://pkg.go.dev/runtime#hdr-Environment_Variables) env variable (it is a soft limit).

- I don't think the logs retention period has direct correlation to the RAM usage (_or at least I'm not experiencing this behavior in Presentator_). The reported 700-800MB RAM usage is probably unused memory caused by occasional spikes due to files upload, thumbs generation, large number of concurrent requests, hooks execution, etc.
The intense disk writes could be caused by the logs though.

- I haven't really done detailed profiling yet so there is probably a lot of places throughout the codebase where performance and allocations could be improved.

**Reply by erickythierry** [reply]
Thanks for the answer. 😁
I'll test it by running the same db on a machine with more limited memory to see the results, but it's already too good for me 😆
